test/unit/forge/routes/api/project_spec.js

```diff
@@ -2330,4 +2330,55 @@ describe('Project API', function () {
             response.statusCode.should.eqls(403)
         })
     })
+    describe('Validate Project Env Vars', function () {
+        it('Reject Duplicate Env Var Names', async function () {
+            const response = await app.inject({
+                method: 'PUT',
+                url: `/api/v1/projects/${app.project.id}`,
+                body: {
+                    settings: {
+                        env: [
+                            { name: 'FOO', value: 'bar' },
+                            { name: 'FOO', value: 'BAR' }
+                        ]
+                    }
+                },
+                cookies: { sid: TestObjects.tokens.alice }
+            })
+            response.should.have.property('statusCode')
+            response.statusCode.should.eqls(400)
+        })
+        it('Reject Invalid names', async function () {
+            const response = await app.inject({
+                method: 'PUT',
+                url: `/api/v1/projects/${app.project.id}`,
+                body: {
+                    settings: {
+                        env: [
+                            { name: '99FOO', value: 'bar' }
+                        ]
+                    }
+                },
+                cookies: { sid: TestObjects.tokens.alice }
+            })
+            response.should.have.property('statusCode')
+            response.statusCode.should.eqls(400)
+        })
+        // it('Reject Illegal names', async function () {
+        //     const response = await app.inject({
+        //         method: 'PUT',
+        //         url: `/api/v1/projects/${app.project.id}`,
+        //         body: {
+        //             settings: {
+        //                 env: [
+        //                     { name: 'FF_FOO', value: 'bar' }
+        //                 ]
+        //             }
+        //         },
+        //         cookies: { sid: TestObjects.tokens.alice }
+        //     })
+        //     response.should.have.property('statusCode')
+        //     response.statusCode.should.eqls(400)
+        // })
```

Comment:
```suggestion
```

Reply:
tempted to leave this along with the commented out test at line 70 in ProjectTemplate to see why rolling back a snapshot is settings `FF_` env vars